lib/IRGen/GenCall.cpp

```diff
@@ -1851,6 +1852,14 @@ static void externalizeArguments(IRGenFunction &IGF, const Callee &callee,
       auto &ti = cast<LoadableTypeInfo>(IGF.getTypeInfo(paramType));
       Address addr = ti.allocateStack(IGF, paramType, false,
                                       "indirect-temporary").getAddress();
+      // Set the alignment the ABI expects.
+      if (AI.getIndirectByVal()) {
+        auto *AS = cast<llvm::AllocaInst>(addr.getAddress());
+        AS->setAlignment(AI.getIndirectAlign().getQuantity());
+        addr = Address(addr.getAddress(),
+                       Alignment(AI.getIndirectAlign().getQuantity()));
```

Comment:
There's no reason not to use the maximum here; IRGen should honor a lower alignment on a loadable type, but who knows, and it will almost certainly generate better code with a higher alignment.